src/components/Input/index.tsx

```diff
@@ -97,30 +157,84 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
       block,
       defaultValue,
       id,
+      onClickCopy = null,
+      shouldShowHideShowButton,
       ...props
     },
     ref
   ) => {
+    const [type, setType] = useState(props.type || 'text')
+    const [wasClicked, setWasClicked] = useState(false)
     const testId = props['data-testid']
     const shouldShowErrorState =
       !(error === false || error === '' || error === undefined) || isUnavailable
     const shouldDisplayInfo = !shouldShowErrorState && info
     const errorMessage = typeof error === 'string' ? error : null
 
+    useEffect(() => {
+      if (wasClicked) {
+        setTimeout(() => {
+          setWasClicked(false)
+        }, 800)
+      }
```

Comment:
it's good hygiene to clean up effects
```suggestion
      if (wasClicked) {
        const timeoutId = setTimeout(() => {
          setWasClicked(false)
        }, 800)
        return () => clearTimeout(timeoutId)
      }
```